Tools/mac-codesign.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+
+# This script signs a specific object with the specified identity, entitlements,
+# and optional flags. If the target is a bundle, it will also sign all frameworks
+# and dylibs within the bundle.
+
+set -eu
+
+function usage() {
+  echo "Usage: $0 [-t] [-e <entitlements file or "preserve">] <identity> <target to codesign>"
+  exit 1
+}
+
+USE_SECURE_TIMESTAMP=0
+ENTITLEMENTS_FILE=""
+
+while getopts ":te:" opt; do
+  case $opt in
+    t)
+      USE_SECURE_TIMESTAMP=1
+      ;;
+    e)
+      ENTITLEMENTS_FILE=$OPTARG
+      ;;
+    \?)
+      usage
+      ;;
+  esac
+done
+
+if [ $USE_SECURE_TIMESTAMP -eq 1 ]; then
+  TIMESTAMP_FLAG="--timestamp"
+else
+  TIMESTAMP_FLAG="--timestamp=none"
+fi
+
+shift $((OPTIND - 1))
+
+if [ $# -ne 2 ]; then
+  usage
+fi
+
+IDENTITY=$1
+TARGET_PATH=$2
+
+# Signs the given target with the specified identity and optional flags.
+function sign() {
+  /usr/bin/codesign -f -s "$IDENTITY" $TIMESTAMP_FLAG ${2:-} "$1"
+}
+
+if [ -d "$TARGET_PATH" ]; then
+  # Newlines are the only valid separator character in find's output.
+  IFS=$'\n'
+
+  for framework in $(find "$TARGET_PATH" -not -path "*/Helpers/*" -name '*.dylib' -or -name '*.framework'); do
+    sign "$framework"
+  done
```

Comment:
I don't think we currently include any of these, but umbrella frameworks include other frameworks inside them. If we end up having one of those at some point I believe this would sign the outer umbrella framework before the inner ones.

Reply:
I changed it so that `-depth` is passed to `find`, which should prioritize deeper items before its parent in case we ever have umbrella frameworks. I hacked up one of these and the changed script signed the child before the parent with this change.